src/App.tsx

```diff
@@ -34,166 +37,178 @@ const App: React.FC = () => {
-      })
-  }
+      )
+      if (response.status === 'ok') {
+        unconfigure()
+      }
+    } catch (error) {
+      console.log('failed unconfigureServer()', error) // eslint-disable-line no-console
+    }
+  }, [unconfigure])
 
-  const fetchElection = async () => {
-    return fetch('/card/read_long')
-      .then((r) => r.json())
-      .then((card) => JSON.parse(card.longValue))
-  }
+  const fetchElection = useCallback(
+    async () =>
+      fetchJSON<CardReadLongResponse>('/card/read_long').then((card) =>
+        JSON.parse(card.longValue)
+      ),
+    []
+  )
 
-  const processCardData = (cardData: CardData, longValueExists: boolean) => {
-    if (cardData.t === 'clerk') {
-      if (!election) {
-        if (longValueExists && !loadingElection) {
-          loadingElection = true
-          fetchElection().then((fetchedElection) => {
-            setElection(fetchedElection)
-            configureServer(fetchedElection)
-            loadingElection = false
-          })
+  const processCardData = useCallback(
+    async (cardData: CardData, longValueExists: boolean) => {
+      if (cardData.t === 'clerk') {
+        if (!election) {
+          if (longValueExists && !loadingElection) {
+            setLoadingElection(true)
+            await uploadElection(await fetchElection())
+            setLoadingElection(false)
+          }
         }
       }
+    },
+    [election, fetchElection, loadingElection, uploadElection]
+  )
+
+  const readCard = useCallback(async () => {
+    try {
+      const card = await fetchJSON<CardReadResponse>('/card/read')
+      if (card.present && card.shortValue) {
+        const cardData = JSON.parse(card.shortValue) as CardData
+        processCardData(cardData, card.longValueExists)
+      }
+    } catch {
+      setCardServerAvailable(false)
     }
-  }
+  }, [processCardData, setCardServerAvailable])
 
-  const readCard = () => {
-    fetch('/card/read')
-      .then((r) => r.json())
-      .then((card) => {
-        if (card.shortValue) {
-          const cardData = JSON.parse(card.shortValue) as CardData
-          processCardData(cardData, card.longValueExists)
-        }
-      })
-      .catch(() => {
-        setCardServerAvailable(false)
+  const scanBatch = useCallback(async () => {
+    try {
+      await fetch('/scan/scanBatch', {
+        method: 'post',
       })
-  }
-
-  const scanBatch = () => {
-    fetch('/scan/scanBatch', {
-      method: 'post',
-    }).catch((error) => {
+    } catch (error) {
       console.log('failed handleFileInput()', error) // eslint-disable-line no-console
-    })
-  }
+    }
+  }, [])
 
-  const invalidateBranch = (event: ButtonEvent) => {
-    const { id } = (event.target as HTMLElement).dataset
-    fetch('/scan/invalidateBatch', {
-      method: 'post',
-      body: id,
-    }).catch((error) => {
+  const invalidateBranch = useCallback(async (event: ButtonEvent) => {
```

Comment:
I imagine this is a holdover typo from before, but is this meant to be `invalidateBatch` ?

Reply:
Looks like it. I'll update it.